.github/workflows/cmake.yml

```diff
@@ -0,0 +1,132 @@
+name: CMake
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+  CCACHE_BASEDIR: ${{github.workspace}}
+  CCACHE_DIR: ${{github.workspace}}/.ccache
+  CCACHE_COMPRESS: true
+  CCACHE_COMPRESSLEVEL: 6
+
+jobs:
+  build-linux:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-20.04, ubuntu-18.04 ]
+      fail-fast: false
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Prepare ccache timestamp
+        id: ccache_cache_timestamp
+        shell: cmake -P {0}
+        run: |
+          string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+          message("::set-output name=timestamp::${current_date}")
+
+      - name: ccache cache files
+        uses: actions/cache@v1.1.0
+        with:
+          path: .ccache
+          key: ${{ runner.os }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
+          restore-keys: |
+            ${{ runner.os }}-ccache-
+
+      - name: Install Ubuntu Dependencies
+        run: >
+            sudo apt install --fix-missing
+            git 
+            ccache 
+            python3-pip 
+            libboost-test-dev 
+            libboost-dev 
+            libboost-filesystem-dev 
+            libboost-program-options-dev 
+            libboost-system-dev 
+            libboost-thread-dev 
+            libgtest-dev 
+            google-mock 
+            libcairo2-dev 
+            libglib2.0-dev 
+            libgtk-3-dev 
+            libtiff5-dev 
+            pkg-config 
+            doxygen 
+            graphviz 
+            libc6-dev
+            
+      - name: Install conan
+        run: pip install conan
+
+      - name: Configure CMake
+        # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a single-configuration generator such as make.
+        # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
+        run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
+
+      - name: Load enviroment
+        run: source ${{github.workspace}}/build/ENVIRONMENT
+
+      - name: Build
+        # Build your program with the given configuration
+        run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}}
+
+#      - name: Test
+#        working-directory: ${{github.workspace}}/build
+#        # Execute tests defined by the CMake configuration.
+#        # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
+#        run: xvfb-run ctest -C ${{env.BUILD_TYPE}}
+
+  build-windows:
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
You guys haven't yet fixed `fetch-depth`

Reply:
Resolved in commit c0de167